### PR TITLE
update onClear and tests

### PR DESCRIPTION
**Pull request by laraharrow**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "labelledby",
     "nomarker",
     "polarbrightness",
+    "seperator",
```
Comment (mdodgelooker) [inline .vscode/settings.json]: Actually spell-check was right – can you update to "separator" in `InputSearchControls` and `MenuSearchControls`?

**Comment by mdodgelooker** [comment]
After chatting with @laraharrow I realized I was totally wrong – `InputSearch` is already controllable, there's even an example of it in `www`. 🙂

File: .vscode/settings.json (hunk 1 above)
Comment (lukelooker) [reply .vscode/settings.json]: This should get removed from this file so folks (like myself) that can't spell don't get caught by this mis-spelling.

**Comment by lukelooker** [comment]
This branch has a bunch of unrelated commits in it – would be good to cut a new branch from latest master and then cherry-pick over the relevant commit:

```
git checkout master
git pull
git checkout -B new-branch-name
git cherry-pick 98ac23f53b3f87b301f18cbe450e48dff7f05006
```

**Pull request closed without merging**